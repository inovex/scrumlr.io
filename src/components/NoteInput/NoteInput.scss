@import "constants/style";

.note-input__icon {
  width: 14px;
  height: 14px;
  fill: var(--accent-color);
  align-self: center;
}
