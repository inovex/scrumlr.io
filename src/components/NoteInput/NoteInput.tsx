import React from "react";
import "./NoteInput.scss";
import {ReactComponent as PlusIcon} from "assets/icon-add.svg";
import store from "store";
import {ActionFactory} from "store/action";
import {InputAdornment, createStyles, withStyles, FilledInput} from "@material-ui/core";

export interface NoteInputProps {
  columnId: string;
}

const CustomInput = withStyles(
  () =>
    createStyles({
      root: {
        borderRadius: 20,
        height: 40,
        width: "100%",
        backgroundColor: "#EDEFF2",
        marginTop: 16,
        marginBottom: 32,
        "&.MuiFilledInput-underline::before , &.MuiFilledInput-underline::after": {
          display: "none",
        },
        "&:hover , &.Mui-focused": {
          backgroundColor: "white",
          boxShadow: "0 6px 9px 0 rgba(0,87,255,0.16)",
        },
      },
      input: {
        color: "black",
        fontSize: 14,
        fontWeight: "bold",
        lineHeight: 24,
        letterSpacing: 0.25,
        padding: "10px 20px 10px 20px",
        // Use the system font instead of the default Roboto font.
        fontFamily: ["Raleway", "sans-serif"].join(","),
        "& .MuiFilledInput-underline:before": {
          borderBottomColor: "green",
          borderBottom: 0,
          position: "relative",
          left: 20,
        },
      },
      "@media (prefers-color-scheme: dark)": {
        root: {
          backgroundColor: "#4C5566",
          "&:hover , &.Mui-focused": {
            backgroundColor: "#4C5566",
            boxShadow: "0 6px 9px 0 #232323",
          },
        },
        input: {
          color: "white",
        },
      },
    }),
  {name: "NoteInput"}
)(FilledInput);

const NoteInput = ({columnId}: NoteInputProps) => {
  const [value, setValue] = React.useState("");

  function handleChangeNotetext(e: React.ChangeEvent<HTMLInputElement>) {
    setValue(e.target.value);
  }
  const onAddNote = () => {
    if (value) {
      store.dispatch(ActionFactory.addNote(columnId!, value));
      setValue("");
    }
  };
  return (
    <CustomInput
      placeholder="Add your note..."
      type="text"
      value={value}
      onChange={handleChangeNotetext}
      onKeyDown={(e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.key === "Enter") {
          onAddNote();
        }
      }}
      endAdornment={
        <InputAdornment position="end">
          <PlusIcon onClick={onAddNote} className="note-input__icon" />
        </InputAdornment>
      }
    />
  );
};
export default NoteInput;
