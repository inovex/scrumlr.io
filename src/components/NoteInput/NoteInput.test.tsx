import React from "react";
import {render} from "@testing-library/react";
import NoteInput from "components/NoteInput/NoteInput";

const createNoteInput = (columnId: string) => <NoteInput columnId={columnId} />;

describe("Note Input", () => {
  beforeEach(() => {
    window.IntersectionObserver = jest.fn(
      () =>
        (({
          observe: jest.fn(),
          disconnect: jest.fn(),
        } as unknown) as IntersectionObserver)
    );
  });

  describe("should render correctly", () => {
    test("noteinput is present", () => {
      const {container} = render(createNoteInput("TestID"));
      expect(container.firstChild).toHaveClass("MuiInputBase-root MuiFilledInput-root NoteInput-root-1 MuiFilledInput-underline MuiInputBase-adornedEnd MuiFilledInput-adornedEnd");
    });
  });

  describe("should have correct style", () => {
    test("show noteinput with correct style", () => {
      const {container} = render(createNoteInput("TestID"));
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
