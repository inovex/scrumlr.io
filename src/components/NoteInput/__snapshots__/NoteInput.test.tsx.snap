// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Input should have correct style show noteinput with correct style 1`] = `
<div
  class="MuiInputBase-root MuiFilledInput-root NoteInput-root-3 MuiFilledInput-underline MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
>
  <input
    class="MuiInputBase-input MuiFilledInput-input NoteInput-input-4 MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
    placeholder="Add your note..."
    type="text"
    value=""
  />
  <div
    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
  >
    <svg
      class="note-input__icon"
    >
      icon-add.svg
    </svg>
  </div>
</div>
`;
